### ISSUE TITLE
Add support for Lambda event filtering for Kinesis Stream

### ISSUE THREAD

**Issue by fwang** [sst/sst#1407]
Waiting for CDK to add support https://github.com/aws/aws-cdk/issues/17874

Here's a workaround
```ts
import { IConstruct } from "constructs";

// Create Kinesis Stream and configure consumers
const stream = new KinesisStream(...);

// Create a function that finds the CloudFormation Lambda EventSourceMapping resource
function findEventSource(c: IConstruct): IConstruct | undefined {
  if (c.cfnResourceType === "AWS::Lambda::EventSourceMapping") {
    return c;
  }

  for (const child of c.node.children) {
    const found = findEventSource(child);
    if (found) {
      return found;
    }
  }
}

// Add event filter directly to the CloudFormation resource
const eventSource = findEventSource(stream);
eventSource?.addPropertyOverride("FilterCriteria", {
  Filters: [
    {
      Pattern: `{ \"eventName\": [\"INSERT\"], \"dynamodb\": { \"NewImage\": {\"channel\": { \"S\" : [\"email\"]}} } }`,
    },
  ],
});
```

Requests:
- Kujtim https://serverless-stack.slack.com/archives/C01JG3B20RY/p1645569793449449
- Jackall https://discord.com/channels/983865673656705025/983866416832864350/993489091268182036

**Comment by Jud** [comment]
Released in CDK 2.41.0 🎉 - https://github.com/aws/aws-cdk/pull/21791

**Comment by thdxr** [comment]
We are transition this repo to https://github.com/sst/v2 and are closing this issue for now. If it is still relevant please reopen it there - sorry for the trouble
